packages/api/src/cli/codegen/languages/typescript/util.ts

```diff
@@ -0,0 +1,74 @@
+import camelCase from 'lodash.camelcase';
+import deburr from 'lodash.deburr';
+import startCase from 'lodash.startcase';
+import { format as prettier } from 'prettier';
+
+export function formatter(content: string) {
+  return prettier(content, {
+    parser: 'typescript',
+    printWidth: 100,
+    singleQuote: true,
+  });
+}
+
+/**
+ * @see {@link https://www.30secondsofcode.org/js/s/word-wrap}
+ */
+export function wordWrap(str: string, max = 88) {
+  return str.replace(new RegExp(`(?![^\\n]{1,${max}}$)([^\\n]{1,${max}})\\s`, 'g'), '$1\n');
+}
+
+/**
+ * Convert a string that might contain spaces or special characters to one that can safely be used
+ * as a TypeScript interface or enum name.
+ *
+ * This function has been adapted and slighty modified from `json-schema-to-typescript`.
+ *
+ * @license MIT
+ * @see {@link https://github.com/bcherny/json-schema-to-typescript}
+ */
+export function toSafeString(str: string) {
+  // identifiers in javaScript/ts:
+  // First character: a-zA-Z | _ | $
+  // Rest: a-zA-Z | _ | $ | 0-9
+
+  // remove accents, umlauts, ... by their basic latin letters
+  return (
+    deburr(str)
+      // if the string starts with a number, prefix it with character that typescript can accept
+      // https://github.com/bcherny/json-schema-to-typescript/issues/489
+      .replace(/^(\d){1}/, '$$1')
```

Comment:
I added these lines to this function in order to fix I did in https://github.com/readmeio/api/pull/530 that resolved type names that started with a number from crashing codegen. If it starts with a number it'll get prefixed with a `$`, which is TS-safe.

Reply:
So wait this is an issue with `json-schema-to-ts` as well?

Reply:
`json-schema-to-ts` doesn't create type names like `json-schema-to-typescript` did, it just gives you the ability to generate type shapes from JSON Schema. We still need to create the names ourselves.